tests/test_non_host.py

```diff
@@ -146,19 +148,10 @@ def test_custom_interpreter_ensure_pythonpath_envar_is_honored(
     fake_metadata = fake_dist / "METADATA"
     with fake_metadata.open("w") as f:
         f.write("Metadata-Version: 2.3\n" "Name: foo\n" "Version: 1.2.3\n")
-    cmd = ["", f"--python={result.creator.exe}"]
+    cmd = ["", f"--python={result.creator.exe}", "--all", "--depth", "0"]
```

Comment:
In some cases, we would get direct dependencies in the `found` variable (i.e. specifically hpy depending on setuptools). I solved this problem by using --all and -d 0 so that all deps are thrown at the "top-level".